shell/app/modules/project/common/components/pipeline-new/form.tsx

```diff
@@ -124,44 +147,61 @@ const PipelineForm = ({ onCancel, pipelineCategory, onOk, appID, fixedApp }: IPr
     };
     if (app.value || fixedApp) {
       initialTree();
-      setTreeValue('');
       setSourceErrorMessage('');
-      form.resetFields(['tree']);
+
+      if (tree.length !== 0) {
+        setTreeValue('');
+        form.resetFields(['tree']);
+      }
     }
   }, [app.value, projectId, getTree, form, fixedApp]);
 
   React.useEffect(() => {
-    if (appID || fixedApp) {
-      form.setFieldsValue?.({ app: appID || fixedApp });
-      setApp({ value: appID || fixedApp });
+    if (fixedApp) {
+      form.setFieldsValue?.({ app: fixedApp });
+      setApp({ value: fixedApp });
+    }
+  }, [form, fixedApp]);
+
+  useEffectOnce(() => {
+    const { id, name, app, fileName, inode } = editData;
+    if (id) {
+      form.setFieldsValue({ name, app, tree: inode });
+      setApp({ value: app as number });
+      setTreeValue(fileName as string);
```

Comment:
app & fileName are `string | undefined` if u r sure this are not undefined, write as `app!` instead